File: backend/src/main/java/com/godeltech/mastery/backend/domain/entity/Equipment.java
package com.godeltech.mastery.backend.domain.entity;

import static javax.persistence.GenerationType.IDENTITY;

import java.io.Serial;
import java.io.Serializable;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
@Table(name = "m_equipments")
public class Equipment implements Serializable {

  @Serial private static final long serialVersionUID = 3251234076822020165L;

  @Id
  @GeneratedValue(strategy = IDENTITY)
  private Long id;

  @Column(name = "name", nullable = false, unique = true)
  private String name;

  @Column(name = "engine_type", nullable = false)
  private String engineType;

  @Column(name = "transmission_type", nullable = false)
  private String transmissionType;

  @Column(name = "engine_size", nullable = false)
  private Double engineSize;

  @Column(name = "power", nullable = false)
  private Integer horsePower;
}

File: backend/src/main/kotlin/com/godeltech/mastery/backend/domain/entity/Equipment.kt
package com.godeltech.mastery.backend.domain.entity

import com.godeltech.mastery.backend.domain.dto.responce.EquipmentDto
import java.io.Serializable
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType.IDENTITY
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "m_equipments")
data class Equipment(

    @Id
    @GeneratedValue(strategy = IDENTITY)
    val id: Long? = null,

    @Column(name = "name")
    var name: String? = null,

    @Column(name = "engine_type")
    var engineType: String? = null,

    @Column(name = "transmission_type")
    var transmissionType: String? = null,

    @Column(name = "engine_size")
    var engineSize: Double? = null,

    @Column(name = "power")
    var horsePower: Int? = null,
) : Serializable {

    fun toDto() = EquipmentDto(
        id = id,
        name = name,
        engineType = engineType,
        transmissionType = transmissionType,
        engineSize = engineSize,
        horsePower = horsePower
    )
}
